boards: dts: nordic: SysCtrl IPC configuration

57300 opened the pull request:
Add DT nodes to be used for communicating with SysCtrl on nRF54H20.

57300 committed:
dts: nordic: nrf54h20: Add SysCtrl VEVIF node

Add a VEVIF node to be used for communicating with SysCtrl (cpusys).
This is the only part of the SysCtrl VPR exposed to local domains.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

57300 committed:
boards: nordic: nrf54h20dk: Add IPC configuration for SysCtrl

Add the default `zephyr,ipc-icmsg` nodes for communication with
Application and Radiocore.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

decsny reviewed (commented):
can you please add dts/common/nordic to the nrf platforms in the maintainer yml, I think this is why kumar was assigned, but it really should be the nordic maintainer assigned

57300 committed:
MAINTAINERS: Update nRF Platforms with new dts paths

Files in `dts/common/nordic` and `dts/riscv/nordic` belong to this area,
now that Nordic's devicetrees span multiple architectures. Use a glob to
cover all paths.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

57300 commented:
> can you please add dts/common/nordic to the nrf platforms in the maintainer yml, I think this is why kumar was assigned, but it really should be the nordic maintainer assigned

Thanks for the suggestion. Updated maintainers and reassigned this PR.

57300 commented:
Rebased and added nodes for Radiocore <-> SysCtrl.

tmon-nordic reviewed (approved):
Should there be some proper error message (instead of cryptic compile error) when `zephyr,ipc-icmsg` compatible (here: `cpuapp_cpusys_ipc`) has status = "okay"; but any of its mboxes (here: `cpuapp_bellboard` and `cpusys_vevif`) have status = "disabled";?

tmon-nordic commented:
> > Should there be some proper error message (instead of cryptic compile error) when `zephyr,ipc-icmsg` compatible (here: `cpuapp_cpusys_ipc`) has status = "okay"; but any of its mboxes (here: `cpuapp_bellboard` and `cpusys_vevif`) have status = "disabled";?
> 
> Sounds reasonable to me. I'm just not familiar enough with Zephyr IPC to know how and where this should be validated, if it isn't already.

It is not validated. I stumbled across the cryptic error at https://github.com/zephyrproject-rtos/zephyr/blob/b4eac67b8b2eee5381d58671fddb8ae9bf2a3989/subsys/ipc/ipc_service/backends/ipc_icmsg.c#L59-L60

Not sure if the `MBOX_DT_SPEC_INST_GET` or `DEFINE_BACKEND_DEVICE` should get the check if the check is to be done at compile level. Alternative approach would be to make devicetree generator to check it. But I don't quite know what would be the correct place.
